sklearn/utils/validation.py

```diff
@@ -2148,9 +2155,10 @@ def _check_sample_weight(
         dtype of the validated `sample_weight`.
         If None, and `sample_weight` is an array:
 
-            - If `sample_weight.dtype` is one of `{np.float64, np.float32}`,
+            - If `sample_weight.dtype` is one of `{xp.float64, xp.float32}`,
               then the dtype is preserved.
-            - Else the output has NumPy's default dtype: `np.float64`.
+            - Otherwise, the output has the highest precision floating point dtype
+              supported by the array namespace/device of the input arrays.
```

Comment:
```suggestion
              supported by the array namespace and device of the input arrays.
```

Reply:
I reverted the docstring changes following https://github.com/scikit-learn/scikit-learn/pull/30878/files#r2085052449.

(I assume your OK with it since you thumbs uped the comment, LMK if you still want the suggestion applied).